pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -499,6 +499,15 @@ public SchemaBuilder addSingleValueDimension(String dimensionName, DataType data
       return this;
     }
 
+    /**
+     * Add single value dimensionFieldSpec with maxLength and a defaultNullValue
+     */
+    public SchemaBuilder addSingleValueDimension(String dimensionName, DataType dataType, int maxLength,
```

Comment:
the maxLength should be applicable only to String type only as discussed earlier. You can add a Precondition Check here to make sure dataType is String.

Reply:
gotcha, added precondition check. thanks for review.